src/nodejs/neo-rpc/v2.9.0/client/module/getNodesBy.js

```diff
@@ -342,3 +342,58 @@ exports.rawmempool = (options) => {
     })
   })
 }
+
+
+// Return an object list of nodes with all stats this module can provide
+// ARGS:
+//  options = {
+//    net: 'TestNet',                                 // default network
+//    nodes: [{ 'url': 'https://host.domain:port' }], // defaults to empty
+//    order: 'asc'                                    // asc = lowest value first, dsc = highest first
+//  }
+// TODO: add caching with configurable time limit for results
+// This will ALWAYS ping first with getNodesBy.ping()
+// Return Value Format:
+// { 'nodeurl': { 'url': url, 'height': height, 'version': version, 'connections': connectionsCount, 'rawmempool': rawmempoolLength}}
+
+exports.all = (options) => {
+  let list = {}, i = 0, j = 0, objCopy, results = {}
+  let getTallest = this.tallest
+  getTallest.keyName = 'height'
+  let getConnections = this.connection
+  getConnections.keyName = 'connections'
+  let getVersion = this.version
+  getVersion.keyName = 'version'
+  let getRawMemPool = this.rawmempool
+  getRawMemPool.keyName = 'rawmempool'
+
+  let operations = [getTallest, getConnections, getVersion, getRawMemPool]
+
+  return new Promise((resolve, reject) => {
+    operations.forEach(op => {
+      op(options).then(rankedNodes => {
+        list[op.keyName] = rankedNodes
+
+        if (i++ === operations.length-1) {
+          operations.forEach(op => {
```

Comment:
`op` is already declared in an upper scope

Reply:
The outermost loop collects the data. The innermost compares it.